Profiler/CProfilerCallback.cpp

```diff
@@ -1,488 +1,555 @@
+	static ShutdownGuard instance;
+	return instance;
+}
+
+void CProfilerCallback::ShutdownFromDllMainDetach() {
+	getShutdownGuard().shutdownInstance(false);
+}
+
+CProfilerCallback::CProfilerCallback() {
+	try {
+		InitializeCriticalSection(&methodSetSynchronization);
+		InitializeCriticalSection(&callbackSynchronization);
+		getShutdownGuard().setInstance(this);
+	}
+	catch (...) {
+		handleException("Constructor");
+	}
+}
+
+CProfilerCallback::~CProfilerCallback() {
+	try {
+		// make sure we flush to disk and disable access to this instance for other threads
+		// even if the .NET framework doesn't call Shutdown() itself
+		getShutdownGuard().shutdownInstance(false);
+		DeleteCriticalSection(&callbackSynchronization);
+		DeleteCriticalSection(&methodSetSynchronization);
+	}
+	catch (...) {
+		handleException("Destructor");
+	}
+}
+
+HRESULT CProfilerCallback::Initialize(IUnknown* pICorProfilerInfoUnkown) {
+	try {
+		return InitializeImplementation(pICorProfilerInfoUnkown);
+	}
+	catch (...) {
+		handleException("Initialize");
+		return S_OK;
+	}
+}
+
+HRESULT CProfilerCallback::InitializeImplementation(IUnknown* pICorProfilerInfoUnkown) {
+	initializeConfig();
+	if (!config.isProfilingEnabled()) {
+		return S_OK;
+	}
+
+	// Place the attach log next to the config and profiler dll
+	std::string configPath = StringUtils::removeLastPartOfPath(config.getConfigPath());
+	attachLog.createLogFile(configPath);
+	attachLog.logAttach();
+
+	traceLog.createLogFile(config.getTargetDir());
+	traceLog.info("looking for configuration options in: " + config.getConfigPath());
+	for (std::string problem : config.getProblems()) {
+		traceLog.error(problem);
+		std::cerr << problem;
+	}
+	if (!config.getProblems().empty()) {
+		WindowsUtils::reportError("Error when loading configuration file", "Couldn't load Profiler.yml configuration for the Teamscale .NET Profiler! See related errors in the standard error stream or in the log file.");
+		// If configuration was incorrect, make it visible to the user by closing the application
+		exit(-1);
+	}
+
+	if (config.shouldUseLightMode()) {
+		traceLog.info("Mode: light");
+	}
+	else {
+		traceLog.info("Mode: force re-jitting");
+	}
+
+	traceLog.info("Eagerness: " + std::to_string(config.getEagerness()));
+
+	if (config.shouldStartUploadDaemon()) {
+		traceLog.info("Starting upload daemon");
+		createDaemon().launch(traceLog);
+	}
+
+	if (config.isTiaEnabled()) {
+		traceLog.info("TIA enabled. REQ Socket: " + config.getTiaRequestSocket());
+		std::function<void(std::string)> testStartCallback = std::bind(&CProfilerCallback::onTestStart, this, std::placeholders::_1);
+		std::function<void(std::string, std::string)> testEndCallback = std::bind(&CProfilerCallback::onTestEnd, this, std::placeholders::_1, std::placeholders::_2);
+		std::function<void(std::string)> errorCallback = std::bind(&TraceLog::error, this->traceLog, std::placeholders::_1);
```

Comment:
Don't use `std::bind` but lambdas for readability (and maybe performance):
```suggestion
std::function<void(std::string)> testStartCallback = [this](std::string testName) {
    this->onTestStart(testName);
};

std::function<void(std::string, std::string)> testEndCallback = [this](std::string result, std::string duration) {
    this->onTestEnd(result, duration);
};

std::function<void(std::string)> errorCallback = [this](std::string message) {
    this->traceLog.error(message);
};
```